app/controllers/games_controller.rb

```diff
@@ -14,17 +13,20 @@ def create
 
   private
 
-  def join_game
-    return if @game.state == 'finished'
-    enlist = Players::Enlist.new(game: @game, player: current_player)
-
-    if enlist.game_full?
-      return render FullGame.template_path
-    else # try to enter the game
-      payload = enlist.assign_game_to_player!
-      set_current_player(enlist.player)
+  def game_params
+    params.require(:game).permit(:name)
+  end
 
-      ActionCable.server.broadcast "game_#{current_player.game_id}", payload
+  def game_template
```

Comment:
This is a very good point. If we want to move this logic out of the controller, **and** this logic is context-aware, **but** we don't want our `Game` model to be context-aware itself, we'll need a third object. I've played around a little, would you mind checking my branch a3ffdf2 and telling me what you think of this alternative approach?